lib/processors/organize-toc.mjs

```diff
@@ -0,0 +1,250 @@
+  }
+
+  // Convert versions to an array, sorted from newest to oldest version.
+  return [...versions].sort(apiVersionSorter).reverse();
+}
+
+// Sorts by API version identifier, from oldest to newest.
+function apiVersionSorter(a, b) {
+  const aMatch = a.match(VERSION_ID_REGEXP_GROUPS);
+  const bMatch = b.match(VERSION_ID_REGEXP_GROUPS);
+
+  // Sort by version number if possible.
+  if (aMatch.groups.version < bMatch.groups.version) {
+    return -1;
+  }
+  if (aMatch.groups.version > bMatch.groups.version) {
+    return 1;
+  }
+
+  // Fall back on launch stage.
+  if (
+    LAUNCH_STAGE_SORT_ORDER.indexOf(aMatch.groups.stage) <
+    LAUNCH_STAGE_SORT_ORDER.indexOf(bMatch.groups.stage)
+  ) {
+    return -1;
+  }
+  if (
+    LAUNCH_STAGE_SORT_ORDER.indexOf(aMatch.groups.stage) >
+    LAUNCH_STAGE_SORT_ORDER.indexOf(bMatch.groups.stage)
+  ) {
+    return 1;
+  }
+
+  // Finally, try the version number associated with the launch stage.
+  if (aMatch.groups.stageVersion < bMatch.groups.stageVersion) {
+    return -1;
+  }
+  if (aMatch.groups.stageVersion > bMatch.groups.stageVersion) {
+    return 1;
+  }
+
+  return 0;
+}
+
+// Sorts lexicographically by the name of each TOC item.
+function tocItemSorter(a, b) {
+  if (a.name < b.name) {
+    return -1;
+  }
+  if (a.name > b.name) {
+    return 1;
+  }
+  return 0;
+}
+
+class TocProcessor {
+  constructor(metadata, obj) {
+    const apiModel = new ApiModel();
+    const apiPackage = apiModel.loadPackage(
+      join(metadata.tmpDir, `${metadata.packageShortName}.api.json`)
+    );
+
+    this._obj = obj;
+    // We normally check the API model to determine which versions are exported.
+    // However, we also allow tests to specify the available versions in the
+    // `metadata` object.
+    this._versions = metadata._versions ?? findExportedVersions(apiPackage);
+
+    // The TOC should have only one main item, which is the library.
+    this._mainItem = obj.items.find(
+      item => item.uid === `${metadata.packageInfo.name}!`
+    );
+    // Remove the main item's children. We'll add them back later, after we
+    // reorganize them. We store them in an `items` property so that we can
+    // recursively iterate over that property.
+    this._items = {
+      items: this._mainItem.items.splice(0),
+    };
+
+    this.noKind = [];
+    Object.keys(TOC_KINDS).forEach(kind => {
+      this[kind] = {
+        // Maps namespaces to the items they contain.
+        namespaces: {},
+        // Items with no namespace.
+        root: [],
+      };
+    });
+  }
+
+  _categorizeItems() {
+    recursiveForEach(this._items, 'items', item => {
+      let kind = item.uid?.match(KIND_REGEXP);
+
+      if (kind) {
+        // Use the captured value as the kind.
+        kind = kind[1];
+      }
+      // Remove ` (Class)` from titles like `FooBar (Class)`.
+      item.name = item.name.replace(TITLE_KIND_REGEXP, '');
+
+      // The package's top-level exported classes get special treatment. To
```

Comment:
I'm a bit unsure on this behavior as we'd only show the newest version's client class at the top-level, and the other versions would be under the `Classes` grouping. I think there's a couple of options we have here:

1. Show all of the top-level classes and add a version to distinguish them e.g. (`ImageAnnotatorClient (v2)` `ImageAnnotatorClient (v1)`).
2. Group top-level classes by version as well, like: 
``` 
  - name: v2
    items:
      - name: ImageAnnotatorClient
        uid: '@google-cloud/vision!v1p4beta1.ImageAnnotatorClient:class'
  - name: v1
    items:
      - name: ImageAnnotatorClient
        uid: '@google-cloud/vision!v1p4beta1.ImageAnnotatorClient:class'
```

What do you think?

Reply:
Sounds good. I went with option 1.